File: sdkwork-backend-sdk-python/sdkwork_backend_sdk/api/user_coupon.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import PlusApiResultBoolean, PlusApiResultListPlusUserCouponVO, PlusApiResultPagePlusUserCouponVO, PlusApiResultPlusUserCouponVO, PlusUserCouponForm, QueryListForm

class UserCouponApi:
    """user_coupon API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def update(self, body: PlusUserCouponForm) -> PlusApiResultPlusUserCouponVO:
        """Update an existing user coupon"""
        return self._client.put(f"/backend/v3/api/user/coupon", json=body)

    def create(self, body: PlusUserCouponForm) -> PlusApiResultPlusUserCouponVO:
        """Create a new user coupon"""
        return self._client.post(f"/backend/v3/api/user/coupon", json=body)

    def list_by_page(self, body: Optional[QueryListForm] = None, params: Optional[Dict[str, Any]] = None) -> PlusApiResultPagePlusUserCouponVO:
        """Get user coupons by page"""
        return self._client.post(f"/backend/v3/api/user/coupon/list", json=body, params=params)

    def list_all_entities(self, body: Optional[QueryListForm] = None) -> PlusApiResultListPlusUserCouponVO:
        """Get all user coupons"""
        return self._client.post(f"/backend/v3/api/user/coupon/list/all", json=body)

    def get_by_id(self, id: str) -> PlusApiResultPlusUserCouponVO:
        """Get a user coupon by ID"""
        return self._client.get(f"/backend/v3/api/user/coupon/{id}")

    def delete(self, id: str) -> PlusApiResultBoolean:
        """Delete a user coupon"""
        return self._client.delete(f"/backend/v3/api/user/coupon/{id}")

File: sdkwork-backend-sdk-flutter/lib/src/api/user_coupon.dart
import '../http/client.dart';
import '../models.dart';

class UserCouponApi {
  final HttpClient _client;
  
  UserCouponApi(this._client);

  /// Update an existing user coupon
  Future<PlusApiResultPlusUserCouponVO?> update(PlusUserCouponForm body) async {
    final response = await _client.put(ApiPaths.backendPath('/user/coupon'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusUserCouponVO ? response : null;
  }

  /// Create a new user coupon
  Future<PlusApiResultPlusUserCouponVO?> create(PlusUserCouponForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/user/coupon'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusUserCouponVO ? response : null;
  }

  /// Get user coupons by page
  Future<PlusApiResultPagePlusUserCouponVO?> listByPage(QueryListForm? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.backendPath('/user/coupon/list'), body: body, params: params, contentType: 'application/json');
    return response is PlusApiResultPagePlusUserCouponVO ? response : null;
  }

  /// Get all user coupons
  Future<PlusApiResultListPlusUserCouponVO?> listAllEntities(QueryListForm? body) async {
    final response = await _client.post(ApiPaths.backendPath('/user/coupon/list/all'), body: body, contentType: 'application/json');
    return response is PlusApiResultListPlusUserCouponVO ? response : null;
  }

  /// Get a user coupon by ID
  Future<PlusApiResultPlusUserCouponVO?> getById(String id) async {
    final response = await _client.get(ApiPaths.backendPath('/user/coupon/${id}'));
    return response is PlusApiResultPlusUserCouponVO ? response : null;
  }

  /// Delete a user coupon
  Future<PlusApiResultBoolean?> delete(String id) async {
    final response = await _client.delete(ApiPaths.backendPath('/user/coupon/${id}'));
    return response is PlusApiResultBoolean ? response : null;
  }
}
